app/workers/retirement_worker.rb

```diff
@@ -15,8 +15,15 @@ def perform(count_id)
 
   def deactivate_workflow!(workflow)
     if workflow.finished?
-      workflow.active = false
-      workflow.save!
+      ignore_optimistic_lock { workflow.update_attribute(:active, false) }
     end
   end
+
+  private
+
+  def ignore_optimistic_lock
+    ActiveRecord::Base.lock_optimistically = false
```

Comment:
I just did a quick proof of concept on this:  https://gist.github.com/parrish/41efa5613077f9dcf818

Basically if you want thread isolated variables, you'll need to set them on the thread.

In the case of an instance variable on ActiveRecord::Base, you might have luck redefining the attribute as a thread variable.  Seems pretty hacky though